src/spider/scheduler/scheduler.cpp

```diff
@@ -38,33 +38,62 @@ constexpr int cCleanupInterval = 5;
 constexpr int cRetryCount = 5;
 
 namespace {
-auto parse_args(int const argc, char** argv) -> boost::program_options::variables_map {
+
+char const* const cUsage
+        = "Usage: spider_scheduler --host <host> --port <port> --storage_url <url>";
+
+auto parse_args(
+        int const argc,
+        char** argv,
+        std::string& host,
+        unsigned short& port,
+        std::string& storage_url
+) -> bool {
     boost::program_options::options_description desc;
-    desc.add_options()("help", "spider scheduler");
-    desc.add_options()(
+    // clang-format off
+    desc.add_options()
+        ("help", "spider scheduler")
+        (
             "host",
```

Comment:
Let's replace these argument names with constants since we're using them in multiple places.